syntax = "proto3";

package todoPackage;

service Todo {
    rpc createTodo(TodoItem) returns (TodoItem);
    rpc readTodos(voidNoParam) returns (TodoItems);
    rpc readTodosStream(voidNoParam) returns (stream TodoItem);

}

message voidNoParam {}

message TodoItem {
    int32 id = 1;
    string details = 2;
    bool done = 3;
}

message TodoItems {
    repeated TodoItem items = 1;
}